main.cpp

```diff
@@ -388,6 +375,44 @@ DoubleType& DoubleType::divide(double x)
     return *this;
 }
 
+//int
+IntType::IntType(int v): value(new int(v)) {}
+
+IntType::~IntType()
+{
+    delete value;
+}
+
+IntType& IntType::pow(int y)
+{
+    powInternal(y);
+    return *this;
+}
+
+IntType& IntType::pow(const FloatType& y)
+{
+    powInternal(static_cast<int>(y));
+    return *this;
+}
+
+IntType& IntType::pow(const DoubleType& y)
+{
+    powInternal(static_cast<int>(y));
+    return *this;
+}
+
+IntType& IntType::pow(const IntType& y)
+{
+    powInternal(static_cast<int>(y));
+    return *this;
+}
```

Comment:
`powInternal` returns `*this` already. 
you don't need a separate return statement.
```
return powInternal(...);
```
- [ ] revise